### PR TITLE
Add debuggertests build

### PR DESCRIPTION
**Pull request by alsemenn**
Re-doing and modifying PR https://github.com/dotnet/coreclr/pull/6696
Comments addressed:
- Having a common script/app for Windows and non-windows platforms - doesn't worth it for now.
- Incremental build flow: no rebuild/copy of the tests needed on the coreclr rebuild.
Comments not addressed:
- Disabling individual debugger tests - needs modifications on debuggertests side

### CODE REVIEW

File: tests/debugger_tests/setup-debuggertets.cmd [hunk 1]
```diff
@@ -0,0 +1,92 @@
+@if not defined __echo @echo off
```
Comment (noahfalk) [inline tests/debugger_tests/setup-debuggertets.cmd]: probably want to name this 'setup-debuggertests.cmd' (there is a missing s in there)

File: tests/debugger_tests/ConfigFilesGenerators/ConfigTemplate.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<Configuration> 
+ <TestRoot>..\..</TestRoot> 
+ <TargetArchitecture>x64</TargetArchitecture> 
+ <ScriptRootDir>$(TestRoot)\Scripts</ScriptRootDir> 
+ <DebuggeeRootDir>$(TestRoot)\Debuggees</DebuggeeRootDir> 
+ <DebuggeeSourceRoot>$(DebuggeeRootDir)\Source</DebuggeeSourceRoot>
+ <RuntimeRoot>##Insert_Runtime_Root##</RuntimeRoot>
+ <NugetCacheRoot>##Insert_Nuget_Cache_Root##</NugetCacheRoot>
+ <Options>
```
Comment (noahfalk) [inline tests/debugger_tests/ConfigFilesGenerators/ConfigTemplate.txt]: You can get rid of the Options and Option tags as there is only 1 option here. It removes a concept that someone would need to understand when trying to figure out what the config file does.

File: tests/debugger_tests/ConfigFilesGenerators/ConfigTemplate.txt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+     <RuntimeSymbolsPath>$(RuntimeRoot)</RuntimeSymbolsPath> 
+     <DebuggeeBuildProcess>cli</DebuggeeBuildProcess> 
+     <CliVersion>1.0.0-preview1-002702</CliVersion> 
+     <DebuggeeNativeLibRoot>$(DebuggeeRootDir)\native</DebuggeeNativeLibRoot> 
+     <DebuggeeBuildRoot>$(DebuggeeRootDir)\dotnet.$(CliVersion)</DebuggeeBuildRoot> 
+     <ProjectJsonNetStandardLibraryVersion>1.0.0-rc2-23811</ProjectJsonNetStandardLibraryVersion> 
+     <ProjectJsonSystemThreadingThreadVersion>4.0.0-rc2-24027</ProjectJsonSystemThreadingThreadVersion> 
+     <ProjectJsonMicrosoftCSharpVersion>4.0.1-rc2-23811</ProjectJsonMicrosoftCSharpVersion> 
+     <ProjectJsonRuntime>win7-x64</ProjectJsonRuntime> 
+     <CliPath>https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/$(CliVersion)/dotnet-dev-win-x64.$(CliVersion).zip</CliPath>
```
Comment (noahfalk) [inline tests/debugger_tests/ConfigFilesGenerators/ConfigTemplate.txt]: Do you want to download a new CLI from the internet? I think coreclr repo already downloaded one as part of its own setup.

**Comment by noahfalk** [comment]
lgtm modulo comments

**Comment by alsemenn** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build

**Pull request merged**